packages/story-editor/src/utils/getMediaBaseColor.js

```diff
@@ -46,16 +47,15 @@ function getImgNodeKey(elementId) {
   return `${IMG_NODE}_${elementId}`;
 }
 
-export function getMediaBaseColor(resource, onBaseColor) {
-  const { type, src, poster } = resource;
-  setOrCreateImage({
-    src: type === 'video' ? poster : src,
+export function getMediaBaseColor(src) {
+  if (!src) {
+    return Promise.reject(new Error('No source to image'));
+  }
+  return setOrCreateImage({
+    src,
     width: 10,
     height: 'auto',
-  }).then(
-    (color) => onBaseColor(color),
-    () => onBaseColor([255, 255, 255]) // Fallback color is white.
-  );
+  });
```

Comment:
Why not `try { ... } catch { return '#ffffff' }` or something?

Reply:
I wanted to return a promise. How you handle a failure, might be different in different usage.

Reply:
That does not mean you cannot return a Promise.

As @miina pointed out, Color Thief returns an error on all-white images. We need to catch that here in `getMediaBaseColor` and return `#ffffff` as a fallback.

It doesn't make sense for every consumer of `getMediaBaseColor` to add this fallback.

It should be done within this function.